src/rust/sqs-executor/src/redis_cache.rs

```diff
@@ -107,26 +118,109 @@ impl RedisCache {
+            let mut lru_cache = self.lru_cache.lock().unwrap();
+
+            cacheables
+                .into_iter()
+                .map(|cacheable| {
+                    // if this hits, it's a Some(CacheResponse::Hit)
+                    // otherwise, we use None to signify that we should check in with redis
+                    let identity_bytes = cacheable.identity();
+                    let lru_cache_response =
+                        lru_cache.get(&identity_bytes).map(|_| CacheResponse::Hit);
+
+                    (cacheable, lru_cache_response)
+                })
+                .collect()
+        };
+
+        // if the LRU cache satisfied all our needs, we should return early
+        let are_all_keys_handled = cacheable_responses
+            .iter()
+            .fold(true, |acc, (_, response)| acc && response.is_some());
+
+        if are_all_keys_handled {
+            return Ok(cacheable_responses
+                .into_iter()
+                .filter_map(|(_, response)| response)
+                .collect());
+        }
+
+        // fetch the responses from redis and convert them into CacheResponses
+        // this Vec<_> should be equal in size to the number of entries in cacheable_responses with
+        // 'response' set to None
+        let mut unknown_cacheable_responses: VecDeque<CacheResponse> = {
+            // this vec will be len() > 0 since we would've returned earlier otherwise
+            let unknown_cacheables: Vec<_> = cacheable_responses
+                .iter()
+                .filter(|(_, response)| response.is_none())
+                .map(|(cacheable, _)| cacheable.identity())
+                .collect();
+
+            let client_pool = self.connection_pool.clone();
+            let mut client = client_pool.get().await;
+
+            client
+                .mget(&unknown_cacheables)
+                .timeout(Duration::from_millis(300))
+                .await??
+                .into_iter()
+                .map(|mget_response: Option<_>| match mget_response {
+                    Some(_) => CacheResponse::Hit,
+                    None => CacheResponse::Miss,
+                })
+                .collect()
+        };
+
+        // convert original cacheable_responses into a complete set of CacheResponses
+        let complete_cacheable_responses: Vec<_> = cacheable_responses
+            .into_iter()
+            .map(|(_, response)| match response {
+                Some(cache_response) => cache_response,
+                None => unknown_cacheable_responses.pop_front().unwrap_or_else(|| {
+                    error!("Missing cacheable response from redis fetch.");
+                    CacheResponse::Miss
+                }),
+            })
+            .collect();
+
+        Ok(complete_cacheable_responses)
+    }
+
     #[tracing::instrument(skip(self, identity))]
     async fn _store(&mut self, identity: Vec<u8>) -> Result<(), RedisCacheError> {
```

Comment:
I realize you didn't do anything with the names of these functions in this PR, but the leading `_` on all of them isn't idiomatic Rust at all. The fact that there's no `pub` modifier will keep it private.

Incidentally, there's also no benefit here to having two separate `impl` blocks for `RedisCache` (one with the public constructor, and the other with all the non-public methods).

Again, I realize these aren't things introduced in this PR... it's just something that stood out as I was reading through it.